src/fragments/lib-v1/datastore/android/data-access/delete-snippet.mdx

```diff
@@ -4,25 +4,34 @@ Below, you query for an instance with an `id` of `"123"`, and then delete it, if
 <Block name="Java">
 
 ```java
-Amplify.DataStore.query(Post.class, Where.id("123"),
-    matches -> {
-        if (matches.hasNext()) {
-            Post post = matches.next();
-            Amplify.DataStore.delete(post,
-                deleted -> Log.i("MyAmplifyApp", "Deleted a post."),
-                failure -> Log.e("MyAmplifyApp", "Delete failed.", failure)
-            );
-        }
-    },
-    failure -> Log.e("MyAmplifyApp", "Query failed.", failure)
-);
+QueryOptions queryOptions = null;
+try {
+    queryOptions = Where.identifier(Post.class, "123");
+} catch (AmplifyException e) {
+    Log.e("MyAmplifyApp", "Failed to construct QueryOptions with provided values for Where.identifier", e);
+}
+
+if (queryOptions != null) {
+    Amplify.DataStore.query(Post.class, queryOptions,
+        matches -> {
+            if (matches.hasNext()) {
+                Post post = matches.next();
+                Amplify.DataStore.delete(post,
+                    deleted -> Log.i("MyAmplifyApp", "Deleted a post."),
+                    failure -> Log.e("MyAmplifyApp", "Delete failed.", failure)
+                );
+            }
+        },
+        failure -> Log.e("MyAmplifyApp", "Query failed.", failure)
+    );
+} 
 ```
 
 </Block>
 <Block name="Kotlin - Callbacks">
 
 ```kotlin
-Amplify.DataStore.query(Post::class.java, Where.id("123"),
+Amplify.DataStore.query(Post::class.java, Where.identifier(Post::class.java, "123"),
```

Comment:
Try/catch block is needed for the Kotlin Callbacks code samples with Where.identifier

Reply:
No compiler warning or error without try/catch block for Kotlin Callbacks